pkg/client/cache/delta_fifo.go

```diff
@@ -522,6 +522,23 @@ func (f *DeltaFIFO) syncKey(key string) error {
 		return nil
 	}
 
+	// If we are doing Resync() and there is already an event queued for that object,
```

Comment:
It is treated as Sync now. But those two are different in fact:
- Sync is supposed to be purely "process my with whatever value I have now"
- Replace is "replace me with that value no matter what value I had before"
Conceptually they are different, but in the code they are treated the same way.